aptos-move/framework/aptos-framework/sources/stake.spec.move

```diff
@@ -0,0 +1,73 @@
+spec aptos_framework::stake {
+    spec calculate_rewards_amount {
+        pragma opaque;
+        ensures [concrete] (rewards_rate_denominator * num_total_proposals == 0) ==> result == 0;
```

Comment:
noob question: What does [concrete] do?

Reply:
"concrete" ones are used to verify the function (`calculate_rewards_amount` in this case). "abstract" ones are used to verify the callers of the function.